source/testing/testing-components.md

```diff
@@ -364,29 +420,35 @@ const locationStub = Service.extend({
   }
 });
 
-moduleForComponent('location-indicator', 'Integration | Component | location indicator', {
-  integration: true,
+module('Integration | Component | location indicator', function(hooks) {
+  setupRenderingTest(hooks);
 
-  beforeEach: function () {
-    this.register('service:location-service', locationStub);
+  hooks.beforeEach(function(assert) {
+    this.owner.register('service:location-service', locationStub);
     // Calling inject puts the service instance in the context of the test,
     // making it accessible as "locationService" within each test
     this.inject.service('location-service', { as: 'locationService' });
-  }
-});
+  });
 
-test('should reveal current location', function(assert) {
-  this.render(hbs`{{location-indicator}}`);
-  assert.equal(this.$().text().trim(), 'You currently are located in New York, USA');
-});
+  test('should reveal current location', function(assert) {
+    await render(hbs`{{location-indicator}}`);
+    assert.equal(this.element.textContent.trim(),
+     'You currently are located in New York, USA');
+  });
+
+  test('should change displayed location when current location changes', async function (assert) {
+    await render(hbs`{{location-indicator}}`);
+
+    assert.equal(this.element.textContent.trim(),
+     'You currently are located in New York, USA', 'origin location should display');
+
+    this.set('locationService.city', 'Beijing');
+    this.set('locationService.country', 'China');
+    this.set('locationService.currentLocation', { x: 11111, y: 222222 });
 
-test('should change displayed location when current location changes', function (assert) {
-  this.render(hbs`{{location-indicator}}`);
-  assert.equal(this.$().text().trim(), 'You currently are located in New York, USA', 'origin location should display');
-  this.set('locationService.city', 'Beijing');
-  this.set('locationService.country', 'China');
-  this.set('locationService.currentLocation', { x: 11111, y: 222222 });
-  assert.equal(this.$().text().trim(), 'You currently are located in Beijing, China', 'location display should change');
+    assert.equal(this.element.textContent.trim(),
+     'You currently are located in Beijing, China', 'location display should change');
+  });
 });
 ```
 
```

Comment:
I wonder if it makes sense to skip the `Waiting on Asynchronous Behavior` section completely, as I feel we'd like to encourage the use of `async / await` over custom helpers in the Guides. Any thoughts?

Reply:
Assuming that async/await covers the same functionality as `wait`, I say we drop the wait explanation. We may still want to cover something in terms of "handling async." It tripped me up as a beginner.

Reply:
A major 👍 on dropping the entire wait explanation below, let's do it. It's been a pain 😀

Reply:
A few small things RE wait. 

I agree that this has been a somewhat difficult thing for us to explain historically,  but now that things are essentially the same between all test types I’m hoping it becomes a bit easier...

 `wait()` has been renamed to `settled()` in @ember/test-helpers (because `await wait()` hurts my eyes). 

Generally speaking, I do think it’s important to review the concept of “settled” and to explain that you may need to `await settled()` to properly pause execution. Also, FWIW nearly all of the helpers discussed here are internally `return settled()`...

Reply:
The "Testing User Interaction" section above (once updated) will already use `async/await` to wait for async effects of user actions so IMHO this section can be dropped

Reply:
> Generally speaking, I do think it’s important to review the concept of “settled” and to explain that you may need to await settled() to properly pause execution. Also, FWIW nearly all of the helpers discussed here are internally return settled()...

I think that's a good point - it might make sense to explain that in this section, too even though the concept of how to use these ready-to-use dom helpers has already been introduced before. I try to update that section accordingly then